props/saber_BC_buttons.h

```diff
@@ -0,0 +1,1092 @@
+  #else
+    // 2 or 3 button
+    case EVENTID(BUTTON_POWER, EVENT_SECOND_SAVED_CLICK_SHORT, MODE_ON | BUTTON_AUX):
+    case EVENTID(BUTTON_POWER, EVENT_SECOND_SAVED_CLICK_SHORT, MODE_OFF | BUTTON_AUX):
+  #endif
+      if (!mode_volume_) {
+        StartOrStopTrack();
+      }
+      return true;
+
+
+// Next Preset AND Volume Up
+
+  #if NUM_BUTTONS == 1
+    case EVENTID(BUTTON_POWER, EVENT_FIRST_CLICK_LONG, MODE_ON):
+  #else
+  // 2 button
+    case EVENTID(BUTTON_POWER, EVENT_FIRST_CLICK_LONG, MODE_ON | BUTTON_AUX):
+  #endif
+    if (fusor.angle1() >  M_PI / 3) {
+      //Don't change preset if in colorchange mode
+      if (SaberBase::GetColorChangeMode() != SaberBase::COLOR_CHANGE_MODE_NONE) return false;
+        next_preset();
+      }
+      return true;
+    case EVENTID(BUTTON_POWER, EVENT_FIRST_CLICK_LONG, MODE_OFF):
+    if (!mode_volume_) {
+      next_preset();
+    } else {
+      VolumeUp();
+    }
+    return true;
+
+
+// Previous Preset AND Volume Down
+
+  #if NUM_BUTTONS == 1
+    case EVENTID(BUTTON_POWER, EVENT_SECOND_CLICK_LONG, MODE_ON):
+  #else
+  // 2 button
+    case EVENTID(BUTTON_POWER, EVENT_SECOND_CLICK_LONG, MODE_ON | BUTTON_AUX):
+  #endif
+    if (fusor.angle1() >  M_PI / 3) {
+      //Don't change preset if in colorchange mode
+      if (SaberBase::GetColorChangeMode() != SaberBase::COLOR_CHANGE_MODE_NONE) return false;
+        previous_preset();
+      }
+      return true;
+    case EVENTID(BUTTON_POWER, EVENT_SECOND_CLICK_LONG, MODE_OFF):
+    if (!mode_volume_) {
+      previous_preset();
+    } else {
+      VolumeDown();
+    }
+    return true;
+
+
+// Enter / Exit Volume MENU
+        
+  #if NUM_BUTTONS == 1
+    case EVENTID(BUTTON_NONE, EVENT_CLASH, MODE_OFF | BUTTON_POWER):
+  #else
+    // 2 button
+    case EVENTID(BUTTON_AUX, EVENT_FIRST_CLICK_LONG, MODE_OFF):
+  #endif
+    if (!mode_volume_) {
+      mode_volume_ = true;
+      if (SFX_vmbegin) {
+        hybrid_font.PlayCommon(&SFX_vmbegin);
+      } else {
+        beeper.Beep(0.5, 3000);
+      }
+      STDOUT.println("Enter Volume Menu");
+    } else {
+      mode_volume_ = false;
+      if (SFX_vmend) {
+        hybrid_font.PlayCommon(&SFX_vmend);
+      } else {
+        beeper.Beep(0.5, 3000);
+      }
+      STDOUT.println("Exit Volume Menu");
+    }
+    return true;
+
+
+// Spoken Battery Level
+        
+  #if NUM_BUTTONS == 1
+    case EVENTID(BUTTON_POWER, EVENT_THIRD_SAVED_CLICK_SHORT, MODE_OFF):
+  #else
+    // 2 and 3 button
+    case EVENTID(BUTTON_AUX, EVENT_FIRST_HELD_LONG, MODE_OFF):
+  #endif
+    // accurate battery level reading every time
+      srand((rand() * 917823) ^ analogRead(batteryLevelPin));
+    // needs this delay
+      delay (500);
+      ::CommandParser::DoParse("play", "battlevel.wav");
+      delay (1900);
+      talkie.SayDigit((int)floorf(battery_monitor.battery()));
+      delay (600);
+      ::CommandParser::DoParse("play", "point.wav");
```

Comment:
Talkie already has a "point" word.
Seems weird to mix wav file playing and talkie playback. Also if we want to do that we need a better way than just callind delay() to do it.

Reply:
Sure, would wait() be better than delay? Not sure how else to make the timing work.

I know it seems weird, but I was just using as much wav as possible for sonic clarity. I tried both ways, and if you actually listen to it it seems more congruent using the point.wav. Either way though, with OS6 pending and the ability to have wavs used for all the digits instead of talkie, would it be moot to even pursue this whole section now?

That being said, I made the wavs conditional on existing.

Reply:
I'm not sure what wait() is, but any function that just sits around and does nothing is not going to work. ProffieOS has better things to do than sitting around and do nothing.

To "wait" for something, you have to build something that polls for it. Usually this means putting something in loop() that checks if it's been long enough to do the next thing.

ProffieOS also have these lightweight threads that can make it look like you can wait for things, but it's all fakery and they end up actually do the same kind of polling that I described above.